karyon3-core/src/test/java/com/netflix/karyon/KaryonTypeBindingFactoryTest.java

```diff
@@ -73,31 +79,90 @@ protected void configure() {
         assertThat(foo.namedBar.getName(),  equalTo("@com.google.inject.name.Named(value=bar)"));
     }
     
-    static class BarAutoBinder implements AutoBinder {
+    @Test
+    public void testPriority() {
+        Injector injector = Karyon.newBuilder()
+            .addModules(new AbstractModule() {
+                @Override
+                protected void configure() {
+                    bind(Foo2.class).asEagerSingleton();
+                }
+            })
+            .addAutoBinder(new BarAutoBinder())
+            .addAutoBinder(new BarAutoBinder2())
+            .start();
+            
+        Foo2 foo = injector.getInstance(Foo2.class);
+        
+        System.out.println("Found: " + foo.bar.getName());
+        assertThat(foo.bar.getName(),       equalTo("-1"));
+    }
+    
+    @Priority(1)
+    static class BarAutoBinder extends AbstractAutoBinder {
+        public BarAutoBinder() {
+            super(KeyMatchers.subclassOf(Bar.class));
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> Module getModuleForKey(final Key<T> key) {
+            return new AbstractModule() {
+                @Override
+                protected void configure() {
+                    bind(key).toInstance((T)new Bar() {
+                        @Override
+                        public String getName() {
+                            return null == key.getAnnotation() ? "" : key.getAnnotation().toString();
+                        } 
+                    });
+                }
+            };
+        }
+    }
+    
+    @Priority(2)
+    static class BarAutoBinder2 extends AbstractAutoBinder {
+        public BarAutoBinder2() {
+            super(KeyMatchers.subclassOf(Bar.class));
+        }
+
         @SuppressWarnings("unchecked")
         @Override
-        public <T> boolean configure(Binder binder, final Key<T> key) {
-            binder.bind(key).toInstance((T)new Bar() {
+        public <T> Module getModuleForKey(final Key<T> key) {
+            return new AbstractModule() {
                 @Override
-                public String getName() {
-                    return null == key.getAnnotation() ? "" : key.getAnnotation().toString();
-                } 
-            });
-            return true;
+                protected void configure() {
+                    bind(key).toInstance((T)new Bar() {
+                        @Override
+                        public String getName() {
+                            return null == key.getAnnotation() ? "-1" : key.getAnnotation().toString() + "-1";
+                        } 
+                    });
+                }
+            };
         }
     }
     
-    static class BazAutoBinder implements AutoBinder {
+    static class BazAutoBinder extends AbstractAutoBinder {
+        public BazAutoBinder() {
+            super(KeyMatchers.subclassOf(Baz.class));
```

Comment:
This seems so common, should we just be calling super(Baz.class) so that the abstract class we distribute could change the reliance on the static KeyMatchers if desired?